common4j/src/main/com/microsoft/identity/common/java/cache/BrokerOAuth2TokenCache.java

```diff
@@ -1638,11 +1642,23 @@ private static <T extends MsalOAuth2TokenCache> T getTokenCache(@NonNull final I
                                                                     @NonNull final INameValueStorage<String> spfm,
                                                                     boolean isFoci) {
         final ICacheKeyValueDelegate cacheKeyValueDelegate = new CacheKeyValueDelegate();
-        final IAccountCredentialCache accountCredentialCache =
-                new SharedPreferencesAccountCredentialCache(
-                        cacheKeyValueDelegate,
-                        spfm
-                );
+        final boolean isFlightEnabled = CommonFlightsManager.INSTANCE
+                .getFlightsProvider()
+                .isFlightEnabled(CommonFlight.USE_IN_MEMORY_CACHE_FOR_ACCOUNTS_AND_CREDENTIALS);
+        final IAccountCredentialCache accountCredentialCache;
+        if (isFlightEnabled) {
+            accountCredentialCache = new SharedPreferencesAccountCredentialCacheWithMemoryCache(
```

Comment:
>SharedPreferencesAccountCredentialCacheWithMemoryCache [](http://example.com/codeflow?start=41&length=54)

Another thing to note, the in-mem cache would keeping creds and account in unencrypted manner. Can you research if it is still secure enough? It maybe, just want to callout and have closure.